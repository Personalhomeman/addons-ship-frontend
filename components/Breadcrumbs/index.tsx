import css from './style.scss';
import SVG from 'react-svg';
import { ReactNode } from 'react';

interface BreadcrumbsProps {
  links: { name: string; url: string }[];
  iconUrl: string;
  children: ReactNode;
}

export default ({ links, iconUrl, children }: BreadcrumbsProps) => (
  <div className={css.breadcrumbs}>
    <div className={css.breadcrumbLinks}>
      {links.map((link, index) => (
        <a key={index} href={link.url} className={css.link}>
          {link.name}
        </a>
      ))}
    </div>
    <div className={css.nameWrapper}>
      <SVG src={iconUrl} className={css.iconUrl} />
      <h2 className={css.name}>{children}</h2>
    </div>
  </div>
);
