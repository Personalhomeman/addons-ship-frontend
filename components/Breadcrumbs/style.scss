@import '~@/assets/style/dimensions';
@import '~@/assets/style/times';
@import '~@/assets/style/colors';

.breadcrumbs {
  .breadcrumbLinks {
    display: flex;

    .link {
      min-width: 0;
      word-wrap: break-word;
      color: $white;
      text-decoration: none;
      // TODO: replace with element from UI library
      line-height: 1.5;
      letter-spacing: normal;
      font-style: normal;
      font-stretch: normal;
      font-size: 19px;
      font-weight: 500;
      opacity: 0.8;
      transition: opacity $medium-hovertime;

      &:before {
        content: 'Â« ';
        white-space: pre;
      }

      &:hover,
      &:focus {
        opacity: 1;
      }

      &:not(:first-child) {
        margin-left: 2 * $grid;
      }
    }
  }

  .nameWrapper {
    display: flex;
    align-items: center;

    .iconUrl {
      margin-right: 2 * $grid;

      svg {
        display: block;
        fill: $white;
      }
    }

    .name {
      min-width: 0;
      word-wrap: break-word;
      color: $white;
      // TODO: replace with element from UI library
      line-height: 1.5;
      letter-spacing: normal;
      font-style: normal;
      font-stretch: normal;
    }
  }
}
