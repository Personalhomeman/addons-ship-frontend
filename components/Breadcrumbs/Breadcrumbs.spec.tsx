import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Breadcrumbs from '.';

describe('Breadcrumbs', () => {
  it('renders correctly', () => {
    const tree = toJSON(shallow(<Breadcrumbs appName='My app' versionIconUrl='https://www.bitrise.io/assets/svg/logo-bitrise.svg'>My app version 1.0</Breadcrumbs>));
    expect(tree).toMatchSnapshot();
  });
});
