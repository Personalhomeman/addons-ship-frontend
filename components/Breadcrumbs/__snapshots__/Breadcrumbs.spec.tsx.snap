// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders correctly 1`] = `
<div
  className="breadcrumbs"
>
  <div
    className="breadcrumbLinks"
  >
    <Link
      href="/"
      key="0"
    >
      <a
        className="link"
      >
        Dashboard
      </a>
    </Link>
    <Link
      href="/my-app/"
      key="1"
    >
      <a
        className="link"
      >
        My app
      </a>
    </Link>
  </div>
  <div
    className="nameWrapper"
  >
    <ReactSVG
      afterInjection={[Function]}
      beforeInjection={[Function]}
      className="iconUrl"
      evalScripts="never"
      fallback={null}
      loading={null}
      renumerateIRIElements={true}
      src="https://www.bitrise.io/assets/svg/logo-bitrise.svg"
      wrapper="div"
    />
    <h2
      className="name"
    >
      My app version 1.0
    </h2>
  </div>
</div>
`;
